server/src/main/java/io/pravega/schemaregistry/server/rest/auth/BasicAuthHandler.java

```diff
@@ -0,0 +1,218 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.server.rest.auth;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import io.pravega.auth.AuthConstants;
+import io.pravega.auth.AuthException;
+import io.pravega.auth.AuthHandler;
+import io.pravega.auth.AuthenticationException;
+import io.pravega.auth.ServerConfig;
+import io.pravega.controller.server.rpc.auth.StrongPasswordProcessor;
+import io.pravega.controller.server.rpc.auth.UserPrincipal;
+import io.pravega.schemaregistry.server.rest.ServiceConfig;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.security.Principal;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.List;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class BasicAuthHandler implements AuthHandler {
```

Comment:
Shouldn't we use Pravega's PasswordAuthHandler here instead of defining a new class?

Reply:
I actually wanted to, but i dont completely recall the trouble i was having with the implementation from controller being plugged in.

Reply:
Also, the implementation in pravega only goes two level deep to check for wild card. 
I wanted a more generic implementation:
a/b/* should be allowed too. 

```          
  // Say, resource is myscope/mystream. ACL specifies permission for myscope/*.
            // Auth should return the the ACL's permissions in that case.
            if (resource.contains("/") && !resource.endsWith("/")) {
                String[] values = resource.split("/");
                if (acl.isResource(values[0] + "/*")) {
                    result = acl.permissions;
                    break;
                }
            }
```

Reply:
We can enhance the one in Pravega to go further levels deep. Please create a Pravega issue for it and also one for Schema Registry to refactor this code later to use the Pravega one.

Reply:
https://github.com/pravega/pravega/issues/4934